apis/v1alpha1/backendpolicy_types.go

```diff
@@ -102,26 +121,28 @@ type BackendTLSConfig struct {
 	// Support: Extended
 	//
 	// +optional
-	CertificateAuthorityRef *LocalObjectReference `json:"certificateAuthorityRef,omitempty"`
+	CertificateAuthorityRef LocalObjectReference `json:"certificateAuthorityRef,omitempty"`
 
 	// Options are a list of key/value pairs to give extended options to the
 	// provider.
 	//
 	// Support: Implementation-specific.
+	//
 	// +optional
 	Options map[string]string `json:"options,omitempty"`
 }
 
 // BackendPolicyStatus defines the observed state of BackendPolicy. Conditions
-// that are related to a specific Route or Gateway should be placed on the
+// that are related to a specific Route or Gateway must be placed on the
 // Route(s) using backends configured by this BackendPolicy.
 type BackendPolicyStatus struct {
 	// Conditions describe the current conditions of the BackendPolicy.
 	//
 	// +listType=map
 	// +listMapKey=type
+	// +kubebuilder:validation:MinItems=1
```

Comment:
Similar to https://github.com/kubernetes-sigs/service-apis/pull/528/files#r571154934. For example:
```
status:
- conditions:
  - type: Admitted 
     status: False
     message: Waiting for controller
     reason: waiting
```

Otherwise we make conditions optional and the lack of conditions means the object has not been reconciled by a controller. I prefer to be explicit instead of requiring the user to infer.

Reply:
Yeh, I agree that there will always be at least one condition.

Reply:
So to clarify, we're saying that status itself is optional, but if it is specified it must include at least one condition? I think status on BackendPolicy is much less clear than other resources. Controllers have clear ownership/management of Gateways and status on Routes is tied directly to the Gateways. That makes it clear which controller is responsible for setting conditions in each case.

Unfortunately with BackendPolicy we don't have a good answer to that. I don't think we ever had a great answer here, but even the godocs for this status suggest that most conditions should be set on route and not here. Given that, I'd rather mark this as optional than require a condition.

Reply:
> So to clarify, we're saying that status itself is optional, but if it is specified it must include at least one condition?

@robscott yes, that is what the changes in this PR propose. I will revert these changes so the PR can proceed.